### PR TITLE
[ktx] Update to v4.3.0-beta1 and fix unix2dos command does not exist

### LINKED ISSUE #35964: [ktx] update to v4.3.0-beta1
### Library name

ktx

### New version number

v4.3.0-beta1

### Other information that may be useful (release notes, etc...)

Upstream fixed https://github.com/microsoft/vcpkg/issues/33687

We can test and close it after port update

**When testing please install ktx[core,tools]:x64-uwp**

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/35964 https://github.com/microsoft/vcpkg/issues/33687

When installing the ktx port in vcpkg, use msys to provide a bash environment. There is no unix2dos command in the bash environment commands provided by msys. So this error was caused. Therefore, the download of unix2dos command was added and the compilation was normal.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x64-windows
x64-uwp
```

### CODE REVIEW

**Comment by jimwang118** [comment]
Submit an issue [834](https://github.com/KhronosGroup/KTX-Software/issues/834) upstream for tracking.

File: ports/ktx/0001-Use-vcpkg-zstd.patch [hunk 1]
```diff
@@ -70,54 +70,52 @@ index 492233a..2663169 100644
...
-@@ -34,9 +34,7 @@ if (EMSCRIPTEN)
+@@ -34,8 +34,7 @@ if (EMSCRIPTEN)
    )
  
    if (KTX2_ZSTANDARD)
 -  	set(SRC_LIST ${SRC_LIST}
 -		../../zstd/zstd.c
--	)
 +  	set(ZSTD_LIBRARIES "$<IF:$<TARGET_EXISTS:zstd::libzstd_shared>,zstd::libzstd_shared,zstd::libzstd_static>")
+ 	)
```
Comment (dg0yt) [inline ports/ktx/0001-Use-vcpkg-zstd.patch:83]: Stray `)`?

File: ports/ktx/0001-Use-vcpkg-zstd.patch [hunk 2]
```diff
@@ -70,54 +70,52 @@ index 492233a..2663169 100644
...
-@@ -28,9 +28,7 @@ if (EMSCRIPTEN)
+@@ -28,8 +28,7 @@ if (EMSCRIPTEN)
    endif()
    
    if (KTX2_ZSTANDARD)
 - 	set(SRC_LIST ${SRC_LIST}
 -		../../zstd/zstddeclib.c
--	)
 + 	set(ZSTD_LIBRARIES "$<IF:$<TARGET_EXISTS:zstd::libzstd_shared>,zstd::libzstd_shared,zstd::libzstd_static>")
+ 	)
```
Comment (dg0yt) [inline ports/ktx/0001-Use-vcpkg-zstd.patch:108]: Stray `)`?

**Comment by dg0yt** [comment]
> there is no unix2dos command in the bash environment commands provided by msys.

It is in the [dos2unix](https://packages.msys2.org/package/dos2unix) package.

File: ports/ktx/portfile.cmake [hunk 3]
```diff
@@ -14,6 +14,14 @@ file(REMOVE "${SOURCE_PATH}/other_include/zstd_errors.h")
...
 vcpkg_list(SET OPTIONS)
 if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_acquire_msys(MSYS_D2U
+        PACKAGES
+            dos2unix
+        DIRECT_PACKAGES
+            # Required for "dos2unix"
+            "https://mirror.msys2.org/msys/x86_64/dos2unix-7.5.1-1-x86_64.pkg.tar.zst"
+            83d85e6ccea746ef9e8153a0d605e774dbe7efc0ee952804acfee4ffd7e3b0386a353b45ff989dd99bc3ce75968209fea3d246ad2af88bbb5c4eca12fc5a8f92
+    )
```
Comment (dg0yt) [inline ports/ktx/portfile.cmake:24]:
```suggestion
```

File: ports/ktx/portfile.cmake [hunk 4]
```diff
@@ -22,6 +30,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
             "https://repo.msys2.org/msys/x86_64/util-linux-2.35.2-3-x86_64.pkg.tar.zst"
             da26540881cd5734072717133307e5d1a27a60468d3656885507833b80f24088c5382eaa0234b30bdd9e8484a6638b4514623f5327f10b19eed36f12158e8edb
```
Comment (dg0yt) [inline ports/ktx/portfile.cmake:31]:
```suggestion
            da26540881cd5734072717133307e5d1a27a60468d3656885507833b80f24088c5382eaa0234b30bdd9e8484a6638b4514623f5327f10b19eed36f12158e8edb
           # Required for "dos2unix"
            "https://mirror.msys2.org/msys/x86_64/dos2unix-7.5.1-1-x86_64.pkg.tar.zst"
            83d85e6ccea746ef9e8153a0d605e774dbe7efc0ee952804acfee4ffd7e3b0386a353b45ff989dd99bc3ce75968209fea3d246ad2af88bbb5c4eca12fc5a8f92
```

File: ports/ktx/portfile.cmake [hunk 5]
```diff
@@ -22,6 +30,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
             "https://repo.msys2.org/msys/x86_64/util-linux-2.35.2-3-x86_64.pkg.tar.zst"
             da26540881cd5734072717133307e5d1a27a60468d3656885507833b80f24088c5382eaa0234b30bdd9e8484a6638b4514623f5327f10b19eed36f12158e8edb
     )
+    file(COPY "${MSYS_D2U}/usr/bin/dos2unix.exe" DESTINATION "${MSYS_ROOT}/usr/bin")
```
Comment (dg0yt) [inline ports/ktx/portfile.cmake:33]:
```suggestion
```

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/ktx/portfile.cmake [hunk 6]
```diff
@@ -42,10 +45,11 @@ vcpkg_cmake_configure(
         -DKTX_FEATURE_LOADTEST_APPS=OFF
         -DKTX_FEATURE_STATIC_LIBRARY=${ENABLE_STATIC}
         ${FEATURE_OPTIONS}
-        ${OPTIONS}
```
Comment (dg0yt) [inline ports/ktx/portfile.cmake:45]: This lines must be restored: It is used to set `BASH_EXECUTABLE`.

File: ports/ktx/portfile.cmake [hunk 7]
```diff
@@ -42,10 +45,11 @@ vcpkg_cmake_configure(
         -DKTX_FEATURE_LOADTEST_APPS=OFF
         -DKTX_FEATURE_STATIC_LIBRARY=${ENABLE_STATIC}
         ${FEATURE_OPTIONS}
-        ${OPTIONS}
         # Do not regenerate headers (needs more dependencies)
         -DCMAKE_DISABLE_FIND_PACKAGE_Vulkan=1
     DISABLE_PARALLEL_CONFIGURE
+    MAYBE_UNUSED_VARIABLES
+        CMAKE_DISABLE_FIND_PACKAGE_Vulkan
```
Comment (dg0yt) [inline ports/ktx/portfile.cmake:52]: I would assume `CMAKE_DISABLE_FIND_PACKAGE_Vulkan` should either not be set, or be controlled by feature `vulkan`.

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**